src/openrct2-ui/interface/ViewportInteraction.cpp

```diff
@@ -637,14 +634,15 @@ static Peep* viewport_interaction_get_closest_peep(ScreenCoordsXY screenCoords,
     screenCoords.x = ((screenCoords.x - viewport->pos.x) * viewport->zoom) + viewport->viewPos.x;
     screenCoords.y = ((screenCoords.y - viewport->pos.y) * viewport->zoom) + viewport->viewPos.y;
 
-    closestPeep = nullptr;
-    closestDistance = 0xFFFF;
-    FOR_ALL_PEEPS (spriteIndex, peep)
+    Peep* closestPeep = nullptr;
+    auto closestDistance = std::numeric_limits<int32_t>::max();
+    auto list = EntityList<Peep>(SPRITE_LIST_PEEP);
+    for (auto peep : list)
```

Comment:
I thought the cpp range for didn't cache the value but that was incorrect. Yes this can be condensed into the loop. I think it pretty much applies for every instance of this.